applications/DEMApplication/custom_elements/thermal_spheric_particle.h

```diff
@@ -1,156 +1,318 @@
+  double mPrescribedHeatFluxSurface;
+  double mPrescribedHeatFluxVolume;
+  double mPrescribedHeatFlux;
+  double mTotalHeatFlux;
+
+  // Delaunay/Voronoi data
+  std::map<int, double> mNeighborVoronoiRadius;
+  int mDelaunayPointListIndex;
+
+  // Turn back information as a string.
+  virtual std::string Info() const override {
+    std::stringstream buffer;
+    buffer << "ThermalSphericParticle";
+    return buffer.str();
+  }
+
+  // Print object information
+  virtual void PrintInfo(std::ostream& rOStream) const override {rOStream << "ThermalSphericParticle";}
+  virtual void PrintData(std::ostream& rOStream) const override {}
+
+  protected:
+
+  // General flags
+  bool is_time_to_solve;      // solve thermal problem in current step
+  bool store_contact_param;   // store contact parameters with neighbors when solving the mechanical problem
+
+  // Neighboring data
+  ThermalSphericParticle<TBaseElement>*     mNeighbor_p;
+  DEMWall*                                  mNeighbor_w;
+  int                                       mNeighborType;
+  unsigned int                              mNeighborIndex;
+  unsigned int                              mNumberOfContactParticleNeighbor;
+  std::map<SphericParticle*, ContactParams> mContactParamsParticle;
+  std::map<DEMWall*, ContactParams>         mContactParamsWall;
+
+  // Interaction properties
+  bool   mNeighborInContact;          // flag for contact interaction
+  double mRealYoungRatio;             // real value of Young modulus
+  double mContactRadius;              // simulation contact radius
+  double mNeighborDistance;           // simulation neighbor distance
+  double mNeighborSeparation;         // simulation neighbor separation (negative value indicates an indentation)
+  double mContactRadiusAdjusted;      // adjusted contact radius from real Young modulus
+  double mNeighborDistanceAdjusted;   // adjusted neighbor distance from adjusted contact radius
+  double mNeighborSeparationAdjusted; // adjusted neighbor separation (negative value indicates an indentation)
+
+  // Radiation environment-related
+  int    mRadiativeNeighbors;
+  double mEnvironmentTemperature;
+  double mEnvironmentTempAux;
+
+  // History-dependent properties
+  double mPreviousTemperature;
+
+  private:
+
+  friend class Serializer;
+
+  virtual void save(Serializer& rSerializer) const override {
+    KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, SphericParticle);
+  }
+
+  virtual void load(Serializer& rSerializer) override {
+    KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, SphericParticle);
+  }
+}; // Class ThermalSphericParticle
+
+  /*
+  // input stream function
+  template<TBaseElement>
+  inline std::istream& operator >> (std::istream& rIStream, ThermalSphericParticle<TBaseElement>& rThis) {return rIStream;}
+
+  // output stream function
+  template<TBaseElement>
+  inline std::ostream& operator << (std::ostream& rOStream, const ThermalSphericParticle<TBaseElement>& rThis)
+  {
```

Comment:
remove commments

Reply:
This commented input/output stream functions are on the other element classes as well, but I dont know how to make it work for a template class.
BTW, after this PR I would like to discuss the possibility of removing the template from the thermal particle, since it can only be a subclass of the spheric particle.